packages/cli/openapi-parser/src/openapi/v3/converters/convertServer.ts

```diff
@@ -3,14 +3,44 @@ import { OpenAPIV3 } from "openapi-types";
 import { getExtension } from "../../../getExtension";
 import { FernOpenAPIExtension } from "../extensions/fernExtensions";
 
+interface ServerConfig {
+    name: string;
+    environment: string | undefined;
+}
```

Comment:
are you up to build this as a zod schema and then instead of `getExtension` we can call `getExtensionAndValidate` and pass the schema in?

Reply:
And then we can make `environment` required (Production/Staging) and `name` optional (`Auth` vs `AI`). This breaks the pattern with `x-fern-server-name` but I think you are moving it in the right direction.